android/debug/AndroidManifest.xml

```diff
@@ -0,0 +1,34 @@
+<!--
+AndroidManifest.xml
+
+Copyright (c) 2017 Couchbase, Inc All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http: www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<manifest
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.couchbase.lite"
+    >
+
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
+
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+
+    <application
+        android:label="@string/app_name"
+        android:usesCleartextTraffic="true"
+        />
+</manifest>
```

Comment:
There is also the same file in `android/androidTest` folder. What is the purpose of this one and the other one in `androidTest` folder?

Reply:
Yeah... that's mysterious.  Android builds the `debug` flavor of the library, for testing.  It then builds an entirely different application that contains the tests.  Both of those applications need to allow clear text traffic, in order for the tests (which use cleartext most fo the time) to pass.